### PR TITLE
feat(snapshot): add `--snapshotId` flag to `push` and `preview` commands

### PR DESCRIPTION
**Pull request by y-lakhdar**
https://coveord.atlassian.net/browse/CDX-514

## Proposed changes

Pushing or previewing a large snapshot leads to a timeout with no possible action.
The idea is to add a `--snapshotId` to both `org:config:push` and `org:config:preview` commands in case the manipulated snapshot leads to a `SnapshotOperationTimeoutError` error.

If that is the case, the user will still be able to preview or push his snapshot once idle.

## Testing

- [ ] Unit Tests:

- [ ] Functionnal Tests:

- [x] Manual Tests: Pushed and preview a large snapshot causing a `SnapshotOperationTimeoutError` error

-----
CDX-514

### CODE REVIEW

File: packages/cli/src/commands/org/config/push.ts [hunk 1]
```diff
@@ -47,6 +49,12 @@ export default class Push extends Command {
       default: false,
       required: false,
     }),
+    snapshotId: flags.string({
+      char: 's',
+      description:
+        'The unique identifier of the snapshot to push. If not specified, a new snapshot will be created from your local project. You can list available snapshot in your organization with org:config:list',
```
Comment (louis-bompart) [inline packages/cli/src/commands/org/config/push.ts:55]:
```suggestion
        'The unique identifier of the snapshot to push. If not specified, a new snapshot will be created from your local project. You can list available snapshots in your organization with org:config:list',
```

Comment (louis-bompart) [reply packages/cli/src/commands/org/config/push.ts:55]: I thought snapshot were not updatable 🤔 
What does pushing to a specific snapshotId does?

**Review by louis-bompart — COMMENTED** [review]
not sure about push

File: packages/cli/src/commands/org/config/push.ts (hunk 1 above)
Comment (y-lakhdar) [reply packages/cli/src/commands/org/config/push.ts]: I am not sure about this one. That's why I wanted your opinion on it.
Would it make sense to have a `--snapshotId` flag on all commands expect for the push?
But like you said, I don't see a lot of use cases expect for someone who wants to apply a snapshot that has been created in the Admin.

Comment (olamothe) [reply packages/cli/src/commands/org/config/push.ts]: I mean... if there's no use case that we see where someone would need to do that (and that we don't even know what happens when you do it), I would certainly not add the flag for push :D 

> Would it make sense to have a --snapshotId flag on all commands expect for the push?

I think that'd be fine, yes

File: packages/cli/src/commands/org/config/push.ts [hunk 2]
```diff
@@ -47,6 +49,12 @@ export default class Push extends Command {
       default: false,
       required: false,
     }),
+    snapshotId: flags.string({
```
Comment (olamothe) [inline packages/cli/src/commands/org/config/push.ts:52]: As said in other comment, I guess we should not have this for `push`.

**Review by olamothe — APPROVED** [review]

**Pull request merged**